### PR TITLE
1951057: Cherry Pick to Add in memory read of cache, delete SCA cert with not needed.

### PR DESCRIPTION
**Pull request by tlhmerry9**

### CODE REVIEW

**Comment by tlhmerry9** [comment]
broke some tests, will fix these

**Comment by jirihnidek** [comment]
Do you really want to cherry-pick anything to the master branch?

**Comment by jirihnidek** [comment]
> Do you really want to cherry-pick anything to the master branch?

I'm sorry. You did it right. The original PR was opened against 1.24 branch: https://github.com/candlepin/subscription-manager/pull/2379. Sorry for confusing you.

**Comment by tlhmerry9** [comment]
no worries, just glad we are all on the same page!  🙂

________________________________

**Comment by tlhmerry9** [comment]
This has 3 commits, 1 for each cherry picked branch.  I can squash them once I get all the tests fixed.

File: src/subscription_manager/managercli.py [hunk 1]
```diff
@@ -53,6 +55,38 @@
 
 log = logging.getLogger(__name__)
 
+conf = config.Config(rhsm.config.initConfig())
+
+SM = "subscription-manager"
+
+
+class SubscribeCommand(AttachCommand):
```
Comment (jirihnidek) [inline src/subscription_manager/managercli.py:63]: Why do you want to add back this alias? It seems unrelated to this PR.

File: src/subscription_manager/managercli.py [hunk 2]
```diff
@@ -53,6 +55,38 @@
...
+        return _("Deprecated, see attach")
+
+    def _command_name(self):
+        return "subscribe"
+
+    def _primary(self):
+        return False
+
+
+class UnSubscribeCommand(RemoveCommand):
```
Comment (jirihnidek) [inline src/subscription_manager/managercli.py:77]: Same comment for this alias?

File: src/subscription_manager/cache.py [hunk 3]
```diff
@@ -1112,3 +1064,51 @@ def _load_data(self, open_file):
         except ValueError:
             # Ignore json file parse error
             pass
+
+
+class ContentAccessModeCache(ConsumerCache):
```
Comment (jirihnidek) [inline src/subscription_manager/cache.py:1069]: Hmm. It looks like that this class is already present in the `main` branch. This only moves it to another place.

**Review by jirihnidek — COMMENTED** [review]
This effort is strange/confusing. It looks like that this fix has already merged to master/main in this PR: https://github.com/candlepin/subscription-manager/pull/2402. It is even more strange, because #2402 is related to this BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1886772 which was opened against RHEL 7.9.

File: src/subscription_manager/managercli.py (hunk 1 above)
Comment (tlhmerry9) [reply src/subscription_manager/managercli.py:63]: Wasn't sure about this so left it knowing it would probably be brought up.  :)  Removed

File: src/subscription_manager/managercli.py (hunk 2 above)
Comment (tlhmerry9) [reply src/subscription_manager/managercli.py:77]: Same for Subscribe/UnSubscribe - wasn't sure so thought I would leave it and let you tell me to remove it.  Removing.

File: src/subscription_manager/cache.py (hunk 3 above)
Comment (tlhmerry9) [reply src/subscription_manager/cache.py]: It was - I was trying to make sure the classes were all in the same order - and I think I was looking at another branch when I made this change.  I put it back to be on line 980 and re-checked all the classes in cache.py and everything looks ok now.

File: test/cli_command_test/test_attach.py [hunk 4]
```diff
@@ -182,3 +182,8 @@ def test_pool_file_invalid(self):
             self.assertEqual(e.code, os.EX_DATAERR)
         else:
             self.fail("No Exception Raised")
+
+
+# Test Attach and Subscribe are the same
+class TestSubscribeCommand(TestAttachCommand):
```
Comment (cnsnyder) [inline test/cli_command_test/test_attach.py:188]: This is unneeded as well, if we've removed the Subscribe command (same for Unsubscribe below)

Comment (tlhmerry9) [reply test/cli_command_test/test_attach.py:188]: True!  :)
